blockio-uring.cabal

```
@@ -73,22 +74,38 @@ benchmark bench
   ghc-options:       -Wall -threaded
 
 test-suite test
+  default-language: Haskell2010
+  type:             exitcode-stdio-1.0
+  hs-source-dirs:   test
+  main-is:          test.hs
+  build-depends:
+    , array
+    , base
+    , blockio-uring
+    , primitive
+    , tasty
+    , tasty-hunit
+    , vector
+
+  ghc-options:      -threaded
+
+test-suite test-internals
   default-language:  Haskell2010
   type:              exitcode-stdio-1.0
-  hs-source-dirs:    test .
-  main-is:           Main.hs
+  hs-source-dirs:    test src
+  main-is:           test-internals.hs
   build-depends:
     , array
     , base
     , primitive
     , tasty
-    , vector
     , tasty-hunit
+    , vector
 
   pkgconfig-depends: liburing
   other-modules:
     System.IO.BlockIO
     System.IO.BlockIO.URing
     System.IO.BlockIO.URingFFI
 
-  ghc-options:       -threaded -fno-ignore-asserts
+  ghc-options:       -threaded
```

It's true that `ghc-options` like `-fno-ignore-asserts` are not transitive across components. But this use here is not transitive. This test suite does not depend on the library component within the package, it includes the sources for the library directly into the test component (`hs-source-dirs: src`), and thus the `ghc-options` takes effect for the library.
```suggestion
  ghc-options:       -threaded -fno-ignore-asserts
```

Of course for the external test suite you've introduced, that depends on the `blockio-uring` lib and doesn't use `src` in its `hs-source-dirs`, so indeed there `-fno-ignore-asserts` would have no useful effect. But for the internal test suite it does have a useful effect (at least potentially, if we have enough tests to exercise the functionality and provoke any assertions).